s32/soc/s32k344/include/Emios_Icu_Ip_Defines.h

```diff
@@ -150,13 +155,29 @@ extern "C"{
 #if ((EMIOS_ICU_IP_EDGE_COUNT_API == STD_ON) || (EMIOS_ICU_IP_SIGNAL_MEASUREMENT_API == STD_ON) || (EMIOS_ICU_IP_TIMESTAMP_API == STD_ON))
 #define EMIOS_ICU_IP_COUNTER_MASK              ((uint32)65535)
 #endif
+
+/* Driver writer needs to define a custom array of indices if,
+ *  Emios Icu channels are used for both PWM SAIC and MCB modes
+ *  as we cannot assume the number channels used
+ */
+#if (DT_HAS_COMPAT_STATUS_OKAY(nxp_qdec_s32) && !(DT_FOREACH_STATUS_OKAY(nxp_s32_emios_pwm, PWM_NXP_S32_CAPTURE_USED)))
+
+#define EMIOS_ICU_IP_INITIAL_INDEX_OF_CHANNELS \
```

Comment:
I don't think is a good idea to limit application by using pwm capture or qdec exclusively.
Beside that seems that you are hardcoding the qdec only uses emios channel 6 and 7,
which is not scalable

Reply:
fixed as per your  suggestions